test/fixtures/job_metric_example/processor.ts

```diff
@@ -0,0 +1,43 @@
+import { DataArray } from './interfaces';
+
+const { BatchProcessor, } = require('@terascope/job-components');
+
+class JobMetricExample extends BatchProcessor {
+    constructor(...args: any) {
+        super(...args);
+        this.initialize();
```

Comment:
You can't call initialize here like this, since initialize is async and constructor can never be async.